### PR TITLE
Change inherit_from configuration to allow authenticate http calls.

### PR DESCRIPTION
**Pull request by ricardoperez**
- Add to the `inherited_decisions ` code ability to make http call with bearer authentication 
  - This would make possible to fetch files from private repositories in github.
- Change `inherit_from` configuration

cli:
```
license_finder inherited_decisions add_with_auth  URL AUTH_TYPE TOKEN_OR_ENV

license_finder inherited_decisions add_with_auth  https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml Bearer HardCodedToken

license_finder inherited_decisions add_with_auth  https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml Bearer \$AUTH_TOKEN
```

Added configuration:
```yml
---
# Get token set in an environment variable called `GIT_BEARER_TOKEN`
- - :inherit_from 
  -
    url: https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml
    authorization: Bearer $GIT_BEARER_TOKEN 

# hardcoded token
- - :inherit_from 
  -  
    url: https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml
    authroization: Bearer HardcodeToekn
```

This would solve issue: #751 
Related PR: #735

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ricardoperez Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=752)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=752) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ricardoperez Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=752)!

File: lib/license_finder/decisions.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def unname_project(txn = {})
 
     def inherit_from(filepath)
       decisions =
-        if filepath =~ %r{^https?://}
+        if filepath.is_a?(Hash)
```
Comment (forelabs) [inline lib/license_finder/decisions.rb:188]: How about keep simply string based url support and add the new hash structure as an addition?

Comment (ricardoperez) [reply lib/license_finder/decisions.rb]: Hi @forelabs , I did that on Friday but forgot to push.

**Comment by ricardoperez** [comment]
> And I think not every host is using Bearer as identifier for the authorization, I know a service using token. So it may be useful to also allow a custom authorization identifier.

Thanks for the feedback. I'll add something to add the authorization identifier.

**Comment by ricardoperez** [comment]
> Also how should I be supposed to add this new format through the cli? The change for the cli is missing here I think.

I haven't think about the cli to be honest, maybe it will need a new command.

**Comment by ricardoperez** [comment]
> I like having this support though I think keep support of the simple format would be nice.
> 
> And I think not every host is using Bearer as identifier for the authorization, I know a service using token. So it may be useful to also allow a custom authorization identifier.
> Also how should I be supposed to add this new format through the cli? The change for the cli is missing here I think.

Hi @forelabs, I've made some changes in the code now is accepting authorization type in the configuration:
```yml
- - :inherit_from 
  -
    url: https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml
    authorization: Bearer $GIT_BEARER_TOKEN 
```

and in the cli:
```
license_finder inherited_decisions add_with_auth  https://raw.githubusercontent.com/repo/master/doc/dependency_decisions.yml Bearer \$GIT_BEARER_TOKEN
```

**Comment by xtreme-shane-lattanzio** [comment]
Hey @ricardoperez! Thanks for the PR. It looks like rubocop is picking up some formatting issues. Can you please correct those and then we can merge this in :)

**Comment by ricardoperez** [comment]
> Hey @ricardoperez! Thanks for the PR. It looks like rubocop is picking up some formatting issues. Can you please correct those and then we can merge this in :)

Fixed.

**Pull request merged**